src/subdomains/core/buy-crypto/process/services/buy-crypto-notification.service.ts

```diff
@@ -65,34 +66,48 @@ export class BuyCryptoNotificationService {
+              : '';
+
+            await this.notificationService.sendMailNew({
               type: MailType.USER,
               input: {
                 userData: tx.user.userData,
-                translationKey: tx.translationKey,
-                translationParams: {
-                  buyInputAmount: tx.inputAmount,
-                  buyInputAsset: tx.inputAsset,
-                  inputBlockchain: tx.cryptoInput ? tx.cryptoInput.asset.blockchain : null,
-                  buyOutputAmount: tx.outputAmount,
-                  buyOutputAsset: tx.outputAsset.name,
-                  blockchain: tx.outputAsset.blockchain,
-                  buyFeePercentage: Util.round(tx.percentFee * 100, 2),
-                  exchangeRate: Util.round(
-                    (tx.inputAmount / tx.inputReferenceAmount) * (tx.inputReferenceAmountMinusFee / tx.outputAmount),
-                    2,
-                  ),
-                  buyWalletAddress: Util.blankStart(tx.target.address),
-                  buyTxId: tx.txId,
-                  buyTransactionLink: tx.transactionId,
-                  fee: `${Util.round(tx.percentFee * 100, 2)}%` + minFee,
+                title: `${MailTranslationKey.BUY_CRYPTO}.title`,
+                prefix: { key: `${MailTranslationKey.BUY_CRYPTO}.salutation` },
+                table: {
+                  [`${MailTranslationKey.BUY_CRYPTO}.input_amount`]: `${tx.inputAmount} ${tx.inputAsset}`,
+                  [`${MailTranslationKey.PAYMENT}.input_blockchain`]: tx.cryptoInput
+                    ? `${tx.cryptoInput.asset.blockchain}`
+                    : null,
+                  [`${MailTranslationKey.BUY_CRYPTO}.output_amount`]: `${tx.outputAmount} ${tx.outputAsset.name}`,
+                  [`${MailTranslationKey.PAYMENT}.blockchain`]: tx.cryptoInput ? null : `${tx.outputAsset.blockchain}`,
+                  [`${MailTranslationKey.PAYMENT}.output_blockchain`]: tx.cryptoInput
+                    ? `${tx.outputAsset.blockchain}`
+                    : null,
+                  [`${MailTranslationKey.PAYMENT}.dfx_fee`]: `${Util.round(tx.percentFee * 100, 2)}%` + minFee,
+                  [`${MailTranslationKey.PAYMENT}.exchange_rate`]: `${tx.exchangeRateString}`,
+                  [`${MailTranslationKey.PAYMENT}.wallet_address`]: Util.blankStart(tx.target.address),
+                  [`${MailTranslationKey.PAYMENT}.transaction_id`]: tx.isLightningOutput
+                    ? Util.blankStart(tx.txId)
+                    : null,
                 },
+                suffix: [
+                  { key: MailKey.SPACE, params: { value: '4' } },
+                  tx.isLightningOutput
+                    ? null
+                    : {
+                        key: `${MailTranslationKey.BUY_CRYPTO}.payment_link`,
+                        params: { url: txExplorerUrl(tx.target.asset.blockchain, tx.txId) },
+                      },
```

Comment:
The space above should only be added, if it is not a lightning output?
(also applies to all other mails)

Reply:
Oh yes good point 👍